pkg/runtime/serializer/codec_factory.go

```diff
@@ -17,36 +17,36 @@ limitations under the License.
 package serializer
 
 import (
+	"io/ioutil"
+
 	"k8s.io/kubernetes/pkg/api/unversioned"
 	"k8s.io/kubernetes/pkg/runtime"
 	"k8s.io/kubernetes/pkg/runtime/serializer/json"
 	"k8s.io/kubernetes/pkg/runtime/serializer/recognizer"
+	"k8s.io/kubernetes/pkg/runtime/serializer/streaming"
 	"k8s.io/kubernetes/pkg/runtime/serializer/versioning"
 )
 
+// serializerExtensions are for serializers that are conditionally compiled in
+var serializerExtensions = []func(*runtime.Scheme) (serializerType, bool){}
+
 type serializerType struct {
 	AcceptContentTypes []string
 	ContentType        string
 	FileExtensions     []string
-	Serializer         runtime.Serializer
-	PrettySerializer   runtime.Serializer
-}
 
-// NewCodecFactory provides methods for retrieving serializers for the supported wire formats
-// and conversion wrappers to define preferred internal and external versions. In the future,
-// as the internal version is used less, callers may instead use a defaulting serializer and
-// only convert objects which are shared internally (Status, common API machinery).
-// TODO: allow other codecs to be compiled in?
-// TODO: accept a scheme interface
-func NewCodecFactory(scheme *runtime.Scheme) CodecFactory {
-	return newCodecFactory(scheme, json.DefaultMetaFactory)
+	Serializer       runtime.Serializer
+	PrettySerializer runtime.Serializer
+	// If set, will be used for streaming codecs. This serializer is assumed to not need explicit
+	// typing via a wrapper.
+	ImplicitSerializer runtime.Serializer
```

Comment:
Got it - thanks for explanation. That makes a lot of sense (and I agree it's better to add it than change the interface of Encode/Decode methods).

Can you extend the comment of that field with roughly what you wrote here?